kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/assertions/nondeterministic/FibonacciIntervalTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.sksamuel.kotest.assertions.nondeterministic
+
+import io.kotest.assertions.nondeterministic.FibonacciIntervalFn
+import io.kotest.assertions.until.FibonacciInterval
+import io.kotest.assertions.until.fibonacci
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.comparables.shouldBeGreaterThan
+import io.kotest.matchers.comparables.shouldBeLessThan
+import io.kotest.matchers.shouldBe
+import kotlin.time.Duration.Companion.minutes
+
+class FibonacciIntervalTest : FunSpec() {
+
+   init {
+      test("fib correctness") {
+         fibonacci(0) shouldBe 0
+         fibonacci(1) shouldBe 1
+         fibonacci(2) shouldBe 1
+         fibonacci(3) shouldBe 2
+         fibonacci(4) shouldBe 3
+         fibonacci(5) shouldBe 5
+         fibonacci(6) shouldBe 8
+         fibonacci(7) shouldBe 13
+      }
+
+      test("fibonacci interval should have a reasonable default max") {
+          val max = FibonacciIntervalFn.defaultMax
+          val default = 10.minutes.fibonacci()
+          val unbounded = 10.minutes.fibonacci(null)
```

Comment:
Using nulls dont make the intent very clear. Wouldn't the same result be achieved by simply providing a really big max duration?

Reply:
Done.